usb/drv.c

```diff
@@ -32,29 +32,54 @@ struct {
 } usbdrv_common;
 
 
-void usb_drvPipeFree(usb_drv_t *drv, usb_pipe_t *pipe)
+
+usb_pipe_t *usb_pipeFind(usb_drv_t *drv, int pipeid)
 {
-	mutexLock(usbdrv_common.lock);
-	if (drv != NULL)
-		idtree_remove(&drv->pipes, &pipe->linkage);
+	usb_pipe_t *res;
```

Comment:
Can pipe be destroyed asynchronously? If so, refcounter might be a good idea, so our pointer to pipe doesn't suddenly become invalid

Reply:
Or is it protected by transfer refcounter? If so it's ok

Reply:
After taking a look at the code I think that giving a reference to pipe pointer outside the locked context in drv.c is not necessary. I will refactor the code so that all the accesses to pipe pointer are done under a this mutex. That would simplify things a lot.